plugins/scaffolder-backend/src/lib/templating/filters.ts

```diff
@@ -15,26 +15,186 @@
  */
 import { parseEntityRef } from '@backstage/catalog-model';
 import { ScmIntegrations } from '@backstage/integration';
-import type { JsonObject, JsonValue } from '@backstage/types';
 import {
+  CreatedTemplateFilter,
+  createTemplateFilter,
   parseRepoUrl,
   TemplateFilter,
 } from '@backstage/plugin-scaffolder-node';
+import type { JsonObject, JsonValue } from '@backstage/types';
+import { keyBy, mapValues } from 'lodash';
 import get from 'lodash/get';
+import { z } from 'zod';
+
+const parseRepoUrlFilter = (integrations: ScmIntegrations) => {
+  return createTemplateFilter({
+    id: 'parseRepoUrl',
+    description:
+      'Parses a repository URL into its components, such as owner, repository name, and more.',
+    schema: {
+      input: z
+        .string()
+        .describe('repo URL as collected from repository picker'),
+      output: z
+        .object({
+          repo: z.string(),
+          host: z.string(),
+        })
+        .merge(
+          z
+            .object({
+              owner: z.string(),
+              organization: z.string(),
+              workspace: z.string(),
+              project: z.string(),
+            })
+            .partial(),
+        )
+        .describe('`RepoSpec`'),
+    },
+    examples: [
+      {
+        example: `- id: log
+name: Parse Repo URL
+action: debug:log
+input:
+extra: \${{ parameters.repoUrl | parseRepoUrl }}`,
+        notes: ` - **Input**: \`github.com?repo=backstage&owner=backstage\`
+- **Output**: \`{"host":"github.com","owner":"backstage","repo":"backstage"}\`
+`,
+      },
+    ],
+    impl: url => parseRepoUrl(url as string, integrations),
+  });
+};
+
+const parseEntityRefFilter = () => {
+  const optionsType = z
+    .object({
+      defaultKind: z
+        .string()
+        .describe('The default kind, if none is given in the reference'),
+      defaultNamespace: z
+        .string()
+        .describe('The default namespace, if none is given in the reference'),
+    })
+    .partial();
 
-export const createDefaultFilters = ({
+  return createTemplateFilter({
```

Comment:
Maybe we can break apart each one of these filters into its own folder, so that we can split apart the examples etc into something a little more readable? It's a lot of code in different formats that makes it tricky to understand.